docs/reference/migration/migrate_6_0/rest.asciidoc

```diff
@@ -8,6 +8,11 @@ This feature was removed in the 5.x series, but a backwards-compatibility layer
 system property `elasticsearch.json.allow_unquoted_field_names`. This backwards-compatibility layer
 has been removed in Elasticsearch 6.0.0.
 
+==== Duplicate Keys in JSON
+
+In previous versions of Elasticsearch, JSON documents were allowed to contain duplicate keys. Elasticsearch 6.0.0
+ enforces that all keys are unique.
```

Comment:
It might be worth writing a blog post around how to migrate documents with these duplicate keys. Because they'll blow up if you try to highlight a field or do source filtering. But they *shouldn't* blow up if you just return the `_source` without filtering.